### DISCUSSION TITLE
How to globally disable the reduction of motion for existing animations?

### DISCUSSION THREAD

**Discussion by pioner92** [software-mansion/react-native-reanimated#5413] [Q&A] [answered]
I would like to disable reduce motions for all animations. 
I have many animation places and I need to add `{reduceMotion:ReduceMotion.Never}` to each place, this is not convenient

**Comment by pioner92** [comment]
here is a fork with this flag disabled
https://github.com/pioner92/react-native-reanimated.git

**Comment by NicholasBoccuzzi** [comment]
This really should be a built-in feature

**Comment by BrodaNoel** [comment]
Agree. Can somebody share a `patch` here?

**Reply by claudesortwell** [reply]
Here is a patch file if anyone needs it, works for both IOS and Android

`react-native-reanimated+3.8.1.patch`
```
diff --git a/node_modules/react-native-reanimated/android/src/main/cpp/NativeProxy.cpp b/node_modules/react-native-reanimated/android/src/main/cpp/NativeProxy.cpp
index e658631..01c0023 100644
--- a/node_modules/react-native-reanimated/android/src/main/cpp/NativeProxy.cpp
+++ b/node_modules/react-native-reanimated/android/src/main/cpp/NativeProxy.cpp
@@ -146,7 +146,7 @@ void NativeProxy::injectCppVersion() {
 void NativeProxy::installJSIBindings() {
   jsi::Runtime &rnRuntime = *rnRuntime_;
   WorkletRuntimeCollector::install(rnRuntime);
-  auto isReducedMotion = getIsReducedMotion();
+  bool isReducedMotion = false;
   RNRuntimeDecorator::decorate(
       rnRuntime, nativeReanimatedModule_, isReducedMotion);
 #ifndef NDEBUG
@@ -172,8 +172,7 @@ void NativeProxy::performOperations() {
 }
 
 bool NativeProxy::getIsReducedMotion() {
-  static const auto method = getJniMethod<jboolean()>("getIsReducedMotion");
-  return method(javaPart_.get());
+  return false;
 }
 
 void NativeProxy::registerNatives() {
diff --git a/node_modules/react-native-reanimated/apple/REAModule.mm b/node_modules/react-native-reanimated/apple/REAModule.mm
index ac42cb8..d721e6b 100644
--- a/node_modules/react-native-reanimated/apple/REAModule.mm
+++ b/node_modules/react-native-reanimated/apple/REAModule.mm
@@ -270,11 +270,13 @@ - (void)sendEventWithName:(NSString *)eventName body:(id)body
     jsi::Runtime &rnRuntime = *jsiRuntime;
     WorkletRuntimeCollector::install(rnRuntime);
 
-#if __has_include(<UIKit/UIAccessibility.h>)
-    auto isReducedMotion = UIAccessibilityIsReduceMotionEnabled();
-#else
-    auto isReducedMotion = NSWorkspace.sharedWorkspace.accessibilityDisplayShouldReduceMotion;
-#endif
+// #if __has_include(<UIKit/UIAccessibility.h>)
+//     auto isReducedMotion = UIAccessibilityIsReduceMotionEnabled();
+// #else
+//     auto isReducedMotion = NSWorkspace.sharedWorkspace.accessibilityDisplayShouldReduceMotion;
+// #endif
+
+    bool isReducedMotion = false;
 
     RNRuntimeDecorator::decorate(rnRuntime, nativeReanimatedModule, isReducedMotion);
 
```

**Comment by MatiPl01** [comment, marked as answer]
Hey!
This feature should be added soon. We already have an [open PR](https://github.com/software-mansion/react-native-reanimated/pull/6164) which will add this functionality.
Hopefully this feature will land in soon. 
I will close this discussion.

**Reply by YOEL311** [reply]
What about this? Now, can I disable the reduction mode once for all app?

**Reply by MatiPl01** [reply]
How would you like to disable reduced motion for all apps? The config affects only the currently running app. We cannot change the behavior of a different app from the currently running app.

If you want to enable/disable reduced motion for all apps, you have to do this in your device settings and then set reduced motion to `ReduceMotion.System` in all apps where you want it to be affected via the device settings.

**Reply by YOEL311** [reply]
It's a typo. I need to disable only my app. I found it.

````
              <ReducedMotionConfig mode={ReduceMotion.Never} />
````

**Reply by MatiPl01** [reply]
Ah, I see. You meant the `entire app`, not `all apps`.

Yeah, this config should disable reduced motion in the entire app. [Here](https://docs.swmansion.com/react-native-reanimated/docs/device/ReducedMotionConfig/) is the respective page in docs with more details about that.
